### DISCUSSION TITLE
Moving files not in subdir when subdirectory-filter

### DISCUSSION THREAD

**Discussion by bill-mccormick-dg** [newren/git-filter-repo#437] [General]
I want to do a --subdirectory-filter on a subdir of the repo root, and keep or move files that are in the repo root that correspond to commits made in the subdir being filtered on. I've tried several variations of --path-rename and --path-rename-match but nothing seems to work. 

For example:
git checkout -b foobranch 
touch foo
touch subdir1/bar
git commit

Then filter
git filter-repo --subdirectory-filter subdir1 --path-rename foo:misc/foo

Should result in the foobranch:
bar
misc/foo

**Comment by newren** [comment]
> git checkout -b foobranch
> touch foo
> touch subdir1/bar
> git commit

You forgot a `git add foo subdir1/bar` step before the git commit.  Without that, the files are not part of history, and filter-repo can only operate on git history.

> git filter-repo --subdirectory-filter subdir1 --path-rename foo:misc/foo

`--subdirectory-filter subdir1` does not mean just rename files under subdir1 to the toplevel, it means to discard any files outside of subdir1 and then rename files under subdir1 to the toplevel.  See the description of that item in the manual:

```
  --subdirectory-filter DIRECTORY
                        Only look at history that touches the given
                        subdirectory and treat that directory as the project
                        root. Equivalent to using '--path DIRECTORY/ --path-
                        rename DIRECTORY/:'
```

If you only want the path rename aspects without the path filtering aspects, then use just that part, i.e. `--path-rename subdir1/:`. 
 So the full command would look something like:

$ git filter-repo --path-rename subdir1/:'' --path-rename foo:misc/foo

> Should result in the foobranch:
> bar
> misc/foo

After making both the above changes, to your example, I do end up with a branch that has those two filenames present.
